mantidproject/mantid
# ConjoinXRuns can receive input workspaces containing Dx values

ConjoinXRuns can now join workspaces which contain Dx values.

- Update of RunCombinationHelper in order to account for Dx value check per spectra. Note this happens during input validation and increases execution time. (used by MergeRuns).

- Add optional Dx to WorkspaceCreationHelper methods
  - Do not overload create2DWorkspaceBinned any more in WorkspaceCreationHelper

- No update of the documentation in terms of usage examples since this functionality (like masking or error propagation) are normal and no highlight of this algorithm. 

Commit messages contain information.


**To test:**

All tests should pass, added unit tests.

Maybe use CreateWorkspace or use CreateSampleWorkspace, add Dx values.

For example:
```Python
values = [1., 2., 3.]
ws = CreateWorkspace(DataX = values, DataY = values, Dx = values)
new_values = [4., 5., 6.]
ws2 = CreateWorkspace(DataX = new_values, DataY = new_values, Dx = new_values)
ws3= ConjoinXRuns([ws, ws2])
```



Fixes #22218 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
